### PR TITLE
fix: structure and naming changes for deleteUsers tests

### PR DESCRIPTION
**Pull request by sanpj2292**
## Description of the change

We have changed the naming convention and updated the folder structure to get data for executing `deleteUsers` tests
The details of the files and their fields have been updated in `/data/deleteUsers/README.md`. Please refer to the same

If `reqParams` are being sent as part of the `http_response.json`, we are validating the parameters with which the http client(in this case `axios`) is being invoked with

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: __tests__/data/deleteUsers/ga/http_response.json [hunk 1]
```diff
@@ -0,0 +1,138 @@
+[
+  [
+    {
+      "type": "post",
+      "reqParams": [
```
Comment (saikumarrs) [inline __tests__/data/deleteUsers/ga/http_response.json:5]: Is there a reason why this is an array and not an object?
That'd give more flexibility in terms of defining this JSON.

Comment (saikumarrs) [reply __tests__/data/deleteUsers/ga/http_response.json:5]: Also, it improves the readability of the data.

File: __tests__/deleteUsers.test.js [hunk 2]
```diff
@@ -21,8 +21,8 @@ const buildPrepareDeleteRequestTestCases = () => {
   .reduce((acc, currentVal) => {
     try {
       const delUsers = require(`../${version}/destinations/${currentVal}/deleteUsers`);
-      const inputJson = require(`./data/${currentVal}_deleteUsers_req_input.json`);
-      const outputJson = require(`./data/${currentVal}_deleteUsers_req_output.json`);
+      const inputJson = require(`./data/deleteUsers/${currentVal}/prepare_req_input.json`);
+      const outputJson = require(`./data/deleteUsers/${currentVal}/prepare_req_output.json`);
```
Comment (saikumarrs) [inline __tests__/deleteUsers.test.js:25]: With the `validateMockAxiosClientReqParams`, do we still need separate req preparation tests?

**Comment by saikumarrs** [comment]
Now that we've touched upon this topic, can we organize the files this way?

```
test
-- genAxios.test.js
-- deleteUsers
   -- deleteUsers.test.js
   -- README.md
   -- data
      -- af
         -- handler_input.json
         -- handler_output.json
      -- ga
         -- handler_input.json
         -- handler_output.json
...
```

File: __tests__/data/deleteUsers/ga/http_response.json (hunk 1 above)
Comment (sanpj2292) [reply __tests__/data/deleteUsers/ga/http_response.json:5]: In the line `expect(mockInstance).toHaveBeenCalledWith(...resp.reqParams);` we are destructuring the array while we are checking if the `axios` instance with relevant arguments. Through this we don't have to worry about how many we are checking with for what type of `axios` instance & all.

I agree about the readability part. But the thing is that the array type implementation makes it easy for us during validation.

File: __mocks__/gen-axios.mock.js [hunk 3]
```diff
@@ -52,4 +58,31 @@ const formAxiosMock = responsesData => {
...
+      break;
+
+    default:
+      mockInstance = axios.post;
+      break;
+  }
+  if (Array.isArray(resp?.reqParams)) {
+    try {
+      expect(mockInstance).toHaveBeenCalled();
+      expect(mockInstance).toHaveBeenCalledWith(...resp.reqParams);
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply __mocks__/gen-axios.mock.js:81]: You're right!

File: __tests__/deleteUsers.test.js (hunk 2 above)
Comment (sanpj2292) [reply __tests__/deleteUsers.test.js:25]: The only reason might help us in catching some corner cases wrt how the request is being prepared. But if we feel they're redundant, we can remove them as well

File: __tests__/data/deleteUsers/README.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
+# DeleteUsers Tests
+
+All the tests data for deleteUsers are to be present in __tests__/data/deleteUsers/${destination}/
+
+### Files and their significance
+  - __`handler_input.json`__ - Input data for  `handleDeletionOfUsers` function in `versionedRouter.js`(alias for `_deleteUsers_proxy_input.json`)
+  - __`handler_output.json`__ - Output of `handleDeletionOfUsers` function in `versionedRouter.js`(alias for `_deleteUsers_proxy_output.json`)
+  - __`prepare_req_input.json`__ - Input data for `prepareDeleteRequest` function exported from the destination's `deleteUsers.js` (An example can be seen in the case of `ga`)
+  - __`prepare_req_output.json`__ - Output of `prepareDeleteRequest` function exported from the destination's `deleteUsers.js` (An example can be seen in the case of `ga`)
+  - __`http_response.json`__ - the mock http responses(An example can be seen in the case of `ga`)
```
[the remark this answers is not in the document]
Comment (sanpj2292) [reply __tests__/data/deleteUsers/README.md]: Done

File: __mocks__/gen-axios.mock.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (sanpj2292) [reply __mocks__/gen-axios.mock.js]: Done

File: __tests__/deleteUsers.test.js (hunk 2 above)
Comment (saikumarrs) [reply __tests__/deleteUsers.test.js]: Let's remove them, as those corner cases also can be tested well now.

**Review by saikumarrs — APPROVED** [review]

File: __tests__/deleteUsers.test.js (hunk 2 above)
Comment (sanpj2292) [reply __tests__/deleteUsers.test.js]: Removed them

**Review by crownedlake — APPROVED** [review]
